registry-prometheus has incorrect dependency versions

### Expected Behavior

Configure prometheus registry:
```yaml
micronaut:
  metrics:
    enabled: true
    tags:
      application: app
    export:
      prometheus:
        enabled: true
        step: PT1M
        descriptions: true
        pushgateway:
          enabled: ${PROMETHEUS_PUSH_GATEWAY_ENABLED}
          address: ${PROMETHEUS_PUSH_GATEWAY_URL}
          interval: ${PROMETHEUS_PUSH_GATEWAY_INTERVAL}
          job: ${PROMETHEUS_PUSH_GATEWAY_JOB_NAME'}
          appName: ${PROMETHEUS_PUSH_GATEWAY_APP_NAME}
```
And have it export the metrics

### Actual Behaviour

```
Exception in thread "main" io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micrometer.core.instrument.composite.CompositeMeterRegistry]

Message: 'boolean io.prometheus.metrics.config.ExporterProperties.getPrometheusTimestampsInMs()'

Caused by: java.lang.NoSuchMethodError: 'boolean io.prometheus.metrics.config.ExporterProperties.getPrometheusTimestampsInMs()'
	at io.prometheus.metrics.expositionformats.ExpositionFormats.init(ExpositionFormats.java:31)
	at io.micrometer.prometheusmetrics.PrometheusMeterRegistry.<init>(PrometheusMeterRegistry.java:107)
	at io.micrometer.prometheusmetrics.PrometheusMeterRegistry.<init>(PrometheusMeterRegistry.java:87)
	at io.micrometer.prometheusmetrics.PrometheusMeterRegistry.<init>(PrometheusMeterRegistry.java:83)
	at io.micronaut.configuration.metrics.micrometer.prometheus.PrometheusMeterRegistryFactory.prometheusConfig(PrometheusMeterRegistryFactory.java:47)
	at io.micronaut.configuration.metrics.micrometer.prometheus.$PrometheusMeterRegistryFactory$PrometheusConfig0$Definition.instantiate(Unknown Source)
```

### Steps To Reproduce

Use the above configuration, and it will fail depending on dependency resolution.

The issue is caused by `micrometer-registry-prometheus` having a requirement on `prometheus-metrics-core:1.2.1`, while also requiring `prometheus-metrics-exposition-formats:1.3.7`
![Image](https://github.com/user-attachments/assets/bb7d3e3a-8122-44a1-a8e8-03e74c021034)

A change happened in `prometheus-metrics-exposition-formats` from 1.3.6 to 1.3.7 that depends on new functionality introduced in `prometheus-metrics-config:1.3.7` (sub-dep of `prometheus-metrics-core`)

### Environment Information

JDK version: 21

### Example Application

_No response_

### Version

4.7.6 - 4.8.2 (tried both)

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -11,7 +11,7 @@ managed-metrics-core = '4.2.30'
 managed-micrometer = '1.13.6'
 managed-micrometer-tracing = '1.4.1'
 managed-new-relic-registry = '0.10.0'
-prometheus-pushgateway = '1.3.6'
+managed-prometheus-metrics-exporter-pushgateway = '1.3.6'
 managed-datasource-micrometer = '1.0.6'
 reflections = '0.10.2'
 
@@ -52,11 +52,11 @@ boms-micrometer = { module = 'io.micrometer:micrometer-bom', version.ref = 'mana
 grpc-api = { module = 'io.grpc:grpc-api' }
 hdr-histogram = { module = 'org.hdrhistogram:HdrHistogram', version.ref = 'hdr-histogram' }
 
-prometheus-metrics-exporter-pushgateway = { module = 'io.prometheus:prometheus-metrics-exporter-pushgateway', version.ref = 'prometheus-pushgateway' }
-prometheus-metrics-config = { module = 'io.prometheus:prometheus-metrics-config', version.ref = 'prometheus-pushgateway' }
-prometheus-metrics-core = { module = 'io.prometheus:prometheus-metrics-core', version.ref = 'prometheus-pushgateway' }
-prometheus-metrics-exposition-formats = { module = 'io.prometheus:prometheus-metrics-exposition-formats', version.ref = 'prometheus-pushgateway' }
-prometheus-metrics-tracer-common = { module = 'io.prometheus:prometheus-metrics-tracer-common', version.ref = 'prometheus-pushgateway' }
+managed-prometheus-metrics-exporter-pushgateway = { module = 'io.prometheus:prometheus-metrics-exporter-pushgateway', version.ref = 'managed-prometheus-metrics-exporter-pushgateway' }
+prometheus-metrics-config = { module = 'io.prometheus:prometheus-metrics-config', version.ref = 'managed-prometheus-metrics-exporter-pushgateway' }
+prometheus-metrics-core = { module = 'io.prometheus:prometheus-metrics-core', version.ref = 'managed-prometheus-metrics-exporter-pushgateway' }
+prometheus-metrics-exposition-formats = { module = 'io.prometheus:prometheus-metrics-exposition-formats', version.ref = 'managed-prometheus-metrics-exporter-pushgateway' }
+prometheus-metrics-tracer-common = { module = 'io.prometheus:prometheus-metrics-tracer-common', version.ref = 'managed-prometheus-metrics-exporter-pushgateway' }
 
 managed-metrics-core = { module = 'io.dropwizard.metrics:metrics-core', version.ref = 'managed-metrics-core' }
 managed-micrometer-registry-new-relic-telemetry = { module = 'com.newrelic.telemetry:micrometer-registry-new-relic', version.ref = 'managed-new-relic-registry' }

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -11,7 +11,7 @@ managed-metrics-core = '4.2.30'
 managed-micrometer = '1.13.6'
 managed-micrometer-tracing = '1.4.1'
 managed-new-relic-registry = '0.10.0'
-managed-prometheus-metrics-exporter-pushgateway = '1.3.6'
+managed-prometheus-pushgateway = '1.3.6'
 managed-datasource-micrometer = '1.0.6'
 reflections = '0.10.2'
 
@@ -52,7 +52,11 @@ boms-micrometer = { module = 'io.micrometer:micrometer-bom', version.ref = 'mana
 grpc-api = { module = 'io.grpc:grpc-api' }
 hdr-histogram = { module = 'org.hdrhistogram:HdrHistogram', version.ref = 'hdr-histogram' }
 
-managed-prometheus-metrics-exporter-pushgateway = { module = 'io.prometheus:prometheus-metrics-exporter-pushgateway', version.ref = 'managed-prometheus-metrics-exporter-pushgateway' }
+managed-prometheus-metrics-exporter-pushgateway = { module = 'io.prometheus:prometheus-metrics-exporter-pushgateway', version.ref = 'managed-prometheus-pushgateway' }
+managed-prometheus-metrics-config = { module = 'io.prometheus:prometheus-metrics-config', version.ref = 'managed-prometheus-pushgateway' }
+managed-prometheus-metrics-core = { module = 'io.prometheus:prometheus-metrics-core', version.ref = 'managed-prometheus-pushgateway' }
+managed-prometheus-metrics-exposition-formats = { module = 'io.prometheus:prometheus-metrics-exposition-formats', version.ref = 'managed-prometheus-pushgateway' }
+managed-prometheus-metrics-tracer-common = { module = 'io.prometheus:prometheus-metrics-tracer-common', version.ref = 'managed-prometheus-pushgateway' }
 
 managed-metrics-core = { module = 'io.dropwizard.metrics:metrics-core', version.ref = 'managed-metrics-core' }
 managed-micrometer-registry-new-relic-telemetry = { module = 'com.newrelic.telemetry:micrometer-registry-new-relic', version.ref = 'managed-new-relic-registry' }

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -11,7 +11,7 @@ managed-metrics-core = '4.2.30'
 managed-micrometer = '1.13.6'
 managed-micrometer-tracing = '1.4.1'
 managed-new-relic-registry = '0.10.0'
-managed-prometheus-pushgateway = '1.3.6'
+prometheus-pushgateway = '1.3.6'
 managed-datasource-micrometer = '1.0.6'
 reflections = '0.10.2'
 
@@ -52,11 +52,11 @@ boms-micrometer = { module = 'io.micrometer:micrometer-bom', version.ref = 'mana
 grpc-api = { module = 'io.grpc:grpc-api' }
 hdr-histogram = { module = 'org.hdrhistogram:HdrHistogram', version.ref = 'hdr-histogram' }
 
-managed-prometheus-metrics-exporter-pushgateway = { module = 'io.prometheus:prometheus-metrics-exporter-pushgateway', version.ref = 'managed-prometheus-pushgateway' }
-managed-prometheus-metrics-config = { module = 'io.prometheus:prometheus-metrics-config', version.ref = 'managed-prometheus-pushgateway' }
-managed-prometheus-metrics-core = { module = 'io.prometheus:prometheus-metrics-core', version.ref = 'managed-prometheus-pushgateway' }
-managed-prometheus-metrics-exposition-formats = { module = 'io.prometheus:prometheus-metrics-exposition-formats', version.ref = 'managed-prometheus-pushgateway' }
-managed-prometheus-metrics-tracer-common = { module = 'io.prometheus:prometheus-metrics-tracer-common', version.ref = 'managed-prometheus-pushgateway' }
+prometheus-metrics-exporter-pushgateway = { module = 'io.prometheus:prometheus-metrics-exporter-pushgateway', version.ref = 'prometheus-pushgateway' }
+prometheus-metrics-config = { module = 'io.prometheus:prometheus-metrics-config', version.ref = 'prometheus-pushgateway' }
+prometheus-metrics-core = { module = 'io.prometheus:prometheus-metrics-core', version.ref = 'prometheus-pushgateway' }
+prometheus-metrics-exposition-formats = { module = 'io.prometheus:prometheus-metrics-exposition-formats', version.ref = 'prometheus-pushgateway' }
+prometheus-metrics-tracer-common = { module = 'io.prometheus:prometheus-metrics-tracer-common', version.ref = 'prometheus-pushgateway' }
 
 managed-metrics-core = { module = 'io.dropwizard.metrics:metrics-core', version.ref = 'managed-metrics-core' }
 managed-micrometer-registry-new-relic-telemetry = { module = 'com.newrelic.telemetry:micrometer-registry-new-relic', version.ref = 'managed-new-relic-registry' }

--- a/micrometer-registry-prometheus-pushgateway/build.gradle
+++ b/micrometer-registry-prometheus-pushgateway/build.gradle
@@ -5,7 +5,7 @@ plugins {
 dependencies {
     api libs.micrometer.registry.prometheus
     api projects.micronautMicrometerRegistryPrometheus
-    implementation libs.prometheus.metrics.exporter.pushgateway
+    implementation libs.managed.prometheus.metrics.exporter.pushgateway
     implementation libs.prometheus.metrics.config
     implementation libs.prometheus.metrics.core
     implementation libs.prometheus.metrics.exposition.formats

--- a/micrometer-registry-prometheus-pushgateway/build.gradle
+++ b/micrometer-registry-prometheus-pushgateway/build.gradle
@@ -5,11 +5,11 @@ plugins {
 dependencies {
     api libs.micrometer.registry.prometheus
     api projects.micronautMicrometerRegistryPrometheus
-    implementation libs.managed.prometheus.metrics.exporter.pushgateway
-    implementation libs.managed.prometheus.metrics.config
-    implementation libs.managed.prometheus.metrics.core
-    implementation libs.managed.prometheus.metrics.exposition.formats
-    implementation libs.managed.prometheus.metrics.tracer.common
+    implementation libs.prometheus.metrics.exporter.pushgateway
+    implementation libs.prometheus.metrics.config
+    implementation libs.prometheus.metrics.core
+    implementation libs.prometheus.metrics.exposition.formats
+    implementation libs.prometheus.metrics.tracer.common
     testImplementation(mnSerde.micronaut.serde.jackson)
 }
 

--- a/micrometer-registry-prometheus-pushgateway/build.gradle
+++ b/micrometer-registry-prometheus-pushgateway/build.gradle
@@ -5,11 +5,11 @@ plugins {
 dependencies {
     api libs.micrometer.registry.prometheus
     api projects.micronautMicrometerRegistryPrometheus
-    implementation libs.managed.prometheus.metrics.exporter.pushgateway
-    implementation libs.prometheus.metrics.config
-    implementation libs.prometheus.metrics.core
-    implementation libs.prometheus.metrics.exposition.formats
-    implementation libs.prometheus.metrics.tracer.common
+    api libs.managed.prometheus.metrics.exporter.pushgateway
+    api libs.prometheus.metrics.config
+    api libs.prometheus.metrics.core
+    api libs.prometheus.metrics.exposition.formats
+    api libs.prometheus.metrics.tracer.common
     testImplementation(mnSerde.micronaut.serde.jackson)
 }
 

--- a/micrometer-registry-prometheus-pushgateway/build.gradle
+++ b/micrometer-registry-prometheus-pushgateway/build.gradle
@@ -5,7 +5,11 @@ plugins {
 dependencies {
     api libs.micrometer.registry.prometheus
     api projects.micronautMicrometerRegistryPrometheus
-    api libs.managed.prometheus.metrics.exporter.pushgateway
+    implementation libs.managed.prometheus.metrics.exporter.pushgateway
+    implementation libs.managed.prometheus.metrics.config
+    implementation libs.managed.prometheus.metrics.core
+    implementation libs.managed.prometheus.metrics.exposition.formats
+    implementation libs.managed.prometheus.metrics.tracer.common
     testImplementation(mnSerde.micronaut.serde.jackson)
 }
 

